packages/esm-ward-app/src/ward-workspace/admit-patient-form-workspace/admit-patient-form.workspace.tsx

```diff
@@ -0,0 +1,259 @@
+import { useInpatientRequest } from '../../hooks/useInpatientRequest';
+import useEmrConfiguration from '../../hooks/useEmrConfiguration';
+import type { AdmitPatientFormWorkspaceProps } from './types';
+
+const AdmitPatientFormWorkspace: React.FC<AdmitPatientFormWorkspaceProps> = ({
+  patient,
+  dispositionType,
+  closeWorkspace,
+  closeWorkspaceWithSavedChanges,
+  promptBeforeClosing,
+}) => {
+  const { t } = useTranslation();
+  const { location } = useWardLocation();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const { mutate: mutateInpatientRequest } = useInpatientRequest();
+  const { emrConfiguration, isLoadingEmrConfiguration, errorFetchingEmrConfiguration } = useEmrConfiguration();
+  const [showErrorNotifications, setShowErrorNotifications] = useState(false);
+  const { isLoading, admissionLocation, mutate: mutateAdmissionLocation } = useAdmissionLocation();
+  const beds = useMemo(() => (isLoading ? [] : filterBeds(admissionLocation)), [admissionLocation]);
+  const isBedManagementModuleInstalled = useFeatureFlag('bedmanagement-module');
+  const getBedRepresentation = useCallback((bedLayout: BedLayout) => {
+    const bedNumber = bedLayout.bedNumber;
+    const patients =
+      bedLayout.patients.length === 0
+        ? [t('emptyText', 'Empty')]
+        : bedLayout.patients.map((patient) => patient?.person?.preferredName?.display);
+    return [bedNumber, ...patients].join(' · ');
+  }, []);
+
+  const zodSchema = useMemo(
+    () =>
+      z.object({
+        bedId: z.number().optional(),
+      }),
+    [isBedManagementModuleInstalled, beds],
+  );
+
+  type FormValues = z.infer<typeof zodSchema>;
+
+  const {
+    control,
+    formState: { errors, isDirty },
+    handleSubmit,
+    getValues,
+    watch,
+  } = useForm<FormValues>({
+    resolver: zodResolver(zodSchema),
+  });
+
+  useEffect(() => {
+    promptBeforeClosing(() => isDirty);
+  }, [isDirty]);
+
+  const onSubmit = useCallback(
+    (values: FormValues) => {
+      setShowErrorNotifications(false);
+      setIsSubmitting(true);
+      const bedSelected = beds.find((bed) => bed.bedId === values.bedId);
+      createEncounter(
+        patient.uuid,
+        dispositionType === 'ADMIT'
+          ? emrConfiguration.admissionEncounterType.uuid
+          : dispositionType === 'TRANSFER'
+            ? emrConfiguration.transferWithinHospitalEncounterType.uuid
+            : null,
+        location?.uuid,
+      )
+        .then(
+          async (response) => {
+            if (response.ok) {
+              if (bedSelected) {
+                return assignPatientToBed(values.bedId, patient.uuid, response.data.uuid);
+              }
+              return response;
+            }
+          },
+          (err: Error) => {
+            showSnackbar({
+              kind: 'error',
+              title: t('errorCreatingEncounter', 'Failed to create {{encounterType}} encounter', {
```

Comment:
Hi @mogoodrich!
Requesting you to please confirm the texts in this file.
Thanks!